pkg/updater/updater.go

```diff
@@ -107,7 +108,8 @@ func GCS(poolCtx context.Context, colClient gcs.Client, groupTimeout, buildTimeo
 		defer cancel()
 		gcsColReader := gcsColumnReader(colClient, buildTimeout, readResult, enableIgnoreSkip)
 		reprocess := 20 * time.Minute // allow 20m for prow to finish uploading artifacts
-		return InflateDropAppend(ctx, log, client, tg, gridPath, write, gcsColReader, reprocess)
+		mets := CreateMetrics(prometheus.NewFactory())
```

Comment:
Ah! So actually, this doesn't need to be created here, the metrics get created in the corresponding 'main.go' file under cmd/, e.g. https://github.com/GoogleCloudPlatform/testgrid/blob/master/cmd/updater/main.go#L168. (I believe you're already good to go and can revert these two lines.)

Reply:
@michelle192837 Understood! just 1 more doubt. Here, we pass `mets` as a param to `InflateDropAppend`, so the return statement needs that `mets` param. What should I pass here?

Reply:
Bubbling it down from update.Update(), refactoring functions along the way, is probably the way to go. So, in this case, adding it to `func GCS(...)` would be my approach.

Reply:
Ah sorry, I missed adding this to InflateDropAppend's call somehow. +1 to what Sean said!

Reply:
@chases2 I don't understand this. 😅

Reply:
Try changing its signature to "func GCS(poolCtx context.Context, colClient gcs.Client, **`mets *Metrics`**, groupTimeout, buildTimeout time.Duration, concurrency int, write bool, enableIgnoreSkip bool) GroupUpdater"

Reply:
Oh I get it now!